daemon/cmd/legacy/daemon_init.go

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package legacy
+
+// marker interface to be able to depend on the legacy daemon
```

Comment:
let's expand the comment to exactly say what part is initialized, e.g. this will make sure the Start hook for Daemon runs before yours and that this means `newDaemon` has been called, but `startDaemon` is likely going to run in parallel with your start hook.

We could just use `struct{}` as the type here instead of `any`, it's perhaps a bit more honest in that there is no information here.

Reply:
makes sense. I changed it accordingly